### PR TITLE
added products page and filters

### PR DESCRIPTION
**Pull request by Hnh013**
added components : product , filter
added context , services for filtering products data 
added page : products page
added route for products page

### CODE REVIEW

File: src/frontend/contexts/dataContext.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { handleAllFilters , initialFilters , productInCartWishlistChecker } from '../services/filtersService';
+import axios from 'axios';
+import { useUser } from './userContext';
+
+const DataContext = createContext();
+
+const DataProvider = ({ children }) => {
+    const [ products , setProducts] = useState([]);
+    const { userState }  = useUser();
+    axios.get('/api/products')
```
Comment (ramalingaa) [inline src/frontend/contexts/dataContext.js:11]: Try using Async await for API calls

**Review by ramalingaa — APPROVED** [review]
Code looks clean, try using updater functions for setState, ex: setState(() => )

**Review by romabulani — APPROVED** [review]
- Code looks clean and modular
- As Ram already mentioned, you should use await in async calls
- Rest all looks good to merge

File: src/frontend/contexts/dataContext.js (hunk 1 above)
Comment (Hnh013) [reply src/frontend/contexts/dataContext.js:11]: sure Rama , will change these to async await calls

**Pull request merged**